### ISSUE TITLE
Synthetic Data from Diffusion Models Improves ImageNet Classification

### ISSUE THREAD

**Issue by shnakazawa** [TeamAidemy/ds-paper-summaries#30]
Azizi, Shekoofeh, et al. “Synthetic Data from Diffusion Models Improves ImageNet Classification.” arXiv [cs.CV], 17 Apr. 2023, http://arxiv.org/abs/2304.08466. arXiv.

- 深層生成モデルで生成したデータを学習に用いることで、画像分類の精度が向上したと報告。
- 生成モデルが data augmentation に使えることを支持する一例。（ただし、あくまで画像分類での話）

## Abstract
> Deep generative models are becoming increasingly powerful, now generating diverse high fidelity photo-realistic samples given text prompts. Have they reached the point where models of natural images can be used for generative data augmentation, helping to improve challenging discriminative tasks? We show that large-scale text-to image diffusion models can be fine-tuned to produce class conditional models with SOTA FID (1.76 at 256x256 resolution) and Inception Score (239 at 256x256). The model also yields a new SOTA in Classification Accuracy Scores (64.96 for 256x256 generative samples, improving to 69.24 for 1024x1024 samples). Augmenting the ImageNet training set with samples from the resulting models yields significant improvements in ImageNet classification accuracy over strong ResNet and Vision Transformer baselines.

(DeepL翻訳)

深層生成モデルはますます強力になってきており、現在では、テキストプロンプトが与えられると、多様で忠実度の高いフォトリアリスティックサンプルを生成する。自然画像のモデルを生成的データ補強に使用し、困難な識別タスクの改善に役立てることができるところまで来ているのだろうか？我々は、SOTAのFID（256x256の解像度で1.76）とInception Score（256x256で239）を持つクラス条件付きモデルを生成するために、大規模なテキストから画像への拡散モデルを微調整できることを示す。このモデルはまた、分類精度スコアにおいて新しいSOTAをもたらします（256x256の生成サンプルで64.96、1024x1024のサンプルで69.24に改善）。得られたモデルのサンプルでImageNetトレーニングセットを補強することで、強力なResNetとVision TransformerのベースラインよりもImageNetの分類精度が大幅に向上します。

## コード

- 本記事作成時点ではコードは公開されていない。

## 解決した課題/先行研究との比較

- 深層生成モデルが作り出す画像のクオリティがどんどん高くなっている。
- **生成モデルはdata augmentationに使えるほど自然な画像を作り出すことができるのだろうか？** を検証した。

## 技術・手法のポイント

- Text-to-Imageの拡散モデル Imagen をImageNetのデータセットでFine-tuning.
- ImageNetに含まれる分類ごとに、分類の割合のバランスを保ちながら画像を生成。
- 生成された画像をImageNetのデータセットに加え、各種分類モデルに学習させた。

## 評価指標

- 生成モデルによるdata augmentation 有り/無しでfine-tuning → 分類タスクを解かせた。
- 分類精度はFrechet Inception Distance (FID) と Inception Score (IS) で評価。
- 結果、ResNet, Transformerベースのモデルでは分類精度の向上が見られた (**Fig.1**)。
[![Image from Gyazo](https://i.gyazo.com/a36f7e2f88fb4128b6af1c7efa3aa526.png)](https://gyazo.com/a36f7e2f88fb4128b6af1c7efa3aa526)
- 生成された画像のサイズが大きいほど、精度改善効果も高い (**Fig.6**)
[![Image from Gyazo](https://i.gyazo.com/2a3d9e29afb13500181e19a3e7878d72.png)](https://gyazo.com/2a3d9e29afb13500181e19a3e7878d72)

- 一方、学習に追加した生成画像の枚数を増やしすぎると精度が低下しだした (**Table 4**)。
[![Image from Gyazo](https://i.gyazo.com/9a194fe08f03fcbed81a408dbfe5b7b4.png)](https://gyazo.com/9a194fe08f03fcbed81a408dbfe5b7b4)

## 残された課題・議論・感想

- 生成画像の枚数を増やしすぎると精度が下がることについては、生成される画像の内容に偏りがあるせいではと議論されている。
	- では、どのような生成画像を作り出せばよいか、生成画像を学習させる方法に工夫ができないかというところが今後の課題。

画像以外のタスクで似たようなことをやるとどうなるのかは興味がある。
元のデータに含まれる画像の特徴にオーバーフィッティングしそうな気もするし、それが大きな問題にならないのは画像分類という多少細部が異なっていても許されることの多いタスクだからではないだろうか。
[Adversarial Random Forest](https://github.com/TeamAidemy/ds-paper-summaries/issues/25) のようなテーブルデータを生成する手法もあるが、本論文の結果を持って「画像分類以外のタスクでも生成モデルがAugmentationに使える」と結論づけるのは早計だろう。（もちろん、手法の限界を正しく理解して使えば問題ないと思う。）

## 重要な引用

- Imagen
	- Saharia, Chitwan, et al. “Photorealistic Text-to-Image Diffusion Models with Deep Language Understanding.” arXiv [cs.CV], 23 May 2022, http://arxiv.org/abs/2205.11487. arXiv.
